src/storage/event_listener.h

```diff
@@ -31,7 +31,10 @@ class EventListener : public rocksdb::EventListener {
   ~EventListener() override = default;
   void OnFlushBegin(rocksdb::DB *db, const rocksdb::FlushJobInfo &fi) override;
   void OnFlushCompleted(rocksdb::DB *db, const rocksdb::FlushJobInfo &fi) override;
+  void OnCompactionBegin(rocksdb::DB *db, const rocksdb::CompactionJobInfo &ci) override;
```

Comment:
I was wondering if we should add functions about Begin, they don't seem to be useful, and with subcompaction-related information, these logs get a lot bigger and don't seem to be very helpful (in my opinion). Actually, we can just keep the functions about Completed, because it contains the `elapsed`, and we'll know when it happened.

Reply:
@caipengbo One case that comes to my mind is a crash or long freeze/lock during the operation. In this case, we can see from logs that the operation had started but wasn't completed. However, I'm not sure if this particular case is common.

Reply:
Maybe that's a case. But if you really need to look at this infomation, perhaps user has been affected (perhaps a write stall has occurred because the compaction is blocking), in which case it might be better to look at the engine log(which contain more information). 

But there's no problem with adding this logging, I just don't think it's useful.

Reply:
I added the Compact Begin log because compaction can take a long time to occur, and compaction can require compacts to confirm the Compact start time while compacts determine the current state.

Reply:
Yes, I think it makes sense to add this for observing the compaction behavior.